tasks/configure.yml

```
@@ -7,3 +7,27 @@
     mode: '0640'
   no_log: true
   with_items: '{{ restic_repos }}'
+
+- name: Check if the repo already exists
```

If you are checkint if repository exists what is the purpose of `init` variable?

---

I'm not completely sure in validity of this kind of check and think it's better not to take any actions on repository unless asked to.

---

Did you check what happens if you initialize same repository twice? If initialization (`restic init`) is idempotent then this kid of check is useless.

---

Just checked and it seems like it is. But removing the check would mean I'll have to ignore errors (it returns a non-zero code) and you won't see an actual error. So the check should stay.

---

That's what `failed_when` option is for. To decide what to fail based not on exit code but on messages it returns.

---

Here is how it works: http://docs.ansible.com/ansible/latest/playbooks_error_handling.html#controlling-what-defines-failure

---

Great! I'll look into that